### ISSUE TITLE
Decay shortcut test failure on Mac OSX Yosemite

### ISSUE THREAD

**Issue by mbmcgarry** [cyclus/cyclus#1089]
Another Yosemite-special error.
```
[ RUN      ] MaterialTest.DecayNever
[       OK ] MaterialTest.DecayNever (20 ms)
[ RUN      ] MaterialTest.DecayShortcut
/Users/mbmcgarry/git/cyclus/tests/material_tests.cc:313: Failure
Value of: m->comp()
  Actual: 0x7fa0fd0464a0
Expected: c
Which is: 0x7fa0fd0463d0
[  FAILED  ] MaterialTest.DecayShortcut (21 ms)
[ RUN      ] MaterialTest.ExtractPrevDecay
[       OK ] MaterialTest.ExtractPrevDecay (23 ms)


[  PASSED  ] 8 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] MaterialTest.DecayShortcut
```

**Comment by rwcarlsen** [comment]
Can @deniadjokic @mbmcgarry try my dbg-decay branch - and paste in results of that here?  It doesn't have the segfault fix - so you will need to `cyclus_unit_test --gtest_filter=Material*`

**Comment by mbmcgarry** [comment]
No failed tests, it looks like this works now too!

Nice debugging, @rwcarlsen.

[==========] Running 25 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from MaterialMatQueryTest
[ RUN      ] MaterialMatQueryTest.ExtractCompleteInexactComp
[       OK ] MaterialMatQueryTest.ExtractCompleteInexactComp (0 ms)
[ RUN      ] MaterialMatQueryTest.ExtractDiffComp
[       OK ] MaterialMatQueryTest.ExtractDiffComp (10 ms)
[----------] 2 tests from MaterialMatQueryTest (10 ms total)

[----------] 23 tests from MaterialTest
[ RUN      ] MaterialTest.Constructors
[       OK ] MaterialTest.Constructors (25 ms)
[ RUN      ] MaterialTest.Clone
[       OK ] MaterialTest.Clone (31 ms)
[ RUN      ] MaterialTest.ExtractRes
[       OK ] MaterialTest.ExtractRes (27 ms)
[ RUN      ] MaterialTest.SimpleAbsorb
[       OK ] MaterialTest.SimpleAbsorb (38 ms)
[ RUN      ] MaterialTest.AbsorbLikeMaterial
[       OK ] MaterialTest.AbsorbLikeMaterial (30 ms)
[ RUN      ] MaterialTest.AbsorbUnlikeMaterial
[       OK ] MaterialTest.AbsorbUnlikeMaterial (29 ms)
[ RUN      ] MaterialTest.AbsorbZeroMaterial
[       OK ] MaterialTest.AbsorbZeroMaterial (27 ms)
[ RUN      ] MaterialTest.AbsorbIntoZeroMaterial
[       OK ] MaterialTest.AbsorbIntoZeroMaterial (30 ms)
[ RUN      ] MaterialTest.ExtractMass
[       OK ] MaterialTest.ExtractMass (32 ms)
[ RUN      ] MaterialTest.ExtractComplete
[       OK ] MaterialTest.ExtractComplete (36 ms)
[ RUN      ] MaterialTest.ExtractOverQty
[       OK ] MaterialTest.ExtractOverQty (29 ms)
[ RUN      ] MaterialTest.ExtractOverComp
[       OK ] MaterialTest.ExtractOverComp (32 ms)
[ RUN      ] MaterialTest.ExtractHalf
[       OK ] MaterialTest.ExtractHalf (30 ms)
[ RUN      ] MaterialTest.AbsorbThenExtract
[       OK ] MaterialTest.AbsorbThenExtract (31 ms)
[ RUN      ] MaterialTest.ExtractInGrams
[       OK ] MaterialTest.ExtractInGrams (33 ms)
[ RUN      ] MaterialTest.DecayManual
[       OK ] MaterialTest.DecayManual (308 ms)
[ RUN      ] MaterialTest.DecayLazy
[       OK ] MaterialTest.DecayLazy (37 ms)
[ RUN      ] MaterialTest.DecayDefault
[       OK ] MaterialTest.DecayDefault (24 ms)
[ RUN      ] MaterialTest.DecayNever
[       OK ] MaterialTest.DecayNever (29 ms)
[ RUN      ] MaterialTest.DecayShortcut
[       OK ] MaterialTest.DecayShortcut (28 ms)
[ RUN      ] MaterialTest.ExtractPrevDecay
[       OK ] MaterialTest.ExtractPrevDecay (28 ms)
[ RUN      ] MaterialTest.TransmutePrevDecay
[       OK ] MaterialTest.TransmutePrevDecay (32 ms)
[ RUN      ] MaterialTest.AbsorbPrevDecay
[       OK ] MaterialTest.AbsorbPrevDecay (24 ms)
[----------] 23 tests from MaterialTest (970 ms total)

[----------] Global test environment tear-down
[==========] 25 tests from 2 test cases ran. (983 ms total)
[  PASSED  ] 25 tests.

**Comment by rwcarlsen** [comment]
I'm not going to PR this until other PRs get in.  Too much rebasing.

**Comment by gidden** [comment]
Can't we cherry-pick the two commits fixing issues on yosemite?

On Tue, Feb 24, 2015 at 12:25 PM, Robert Carlsen <notifications@github.com>
wrote:

> I'm not going to PR this until other PRs get in. Too much rebasing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/issues/1089#issuecomment-75814892>.
>



-- 
Matthew Gidden
Ph.D. Candidate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by rwcarlsen** [comment]
If debugging others' problems just gets me lots more rebasing work for my own problems... seems like a perverse incentive.  And - collateral ;-)

**Comment by deniadjokic** [comment]
All tests passed for me too. Thank you, @rwcarlsen !

**Comment by scopatz** [comment]
closed by #1091

**Comment by scopatz** [comment]
Woohoo!
